src/frontend/state/OverlayState.ts

```diff
@@ -0,0 +1,85 @@
+import { OverlayStateInterface } from 'common/state/interfaces'
+import { OverlayRenderState } from 'common/types'
+import { makeAutoObservable } from 'mobx'
+
+class OverlayState implements OverlayStateInterface {
+  renderState: OverlayRenderState = {
+    showToasts: false,
+    showBrowserGame: false,
+    browserGameUrl: '',
+    showHintText: false,
+    showExitGameButton: false,
+    showExtension: false,
+    showBackgroundTint: false
+  }
+
+  showOverlay = false
+
+  isFullscreenOverlay = false
+
+  title = ''
+
+  constructor() {
+    makeAutoObservable(
+      this,
+      {},
+      {
+        deep: true,
+        proxy: false,
```

Comment:
it makes it easier to debug, a little bit faster and the shape of the OverlayState object doesn't change over time

Reply:
https://mobx.js.org/observable-state.html#avoid-proxies

```
The second difference is that observable creates a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) object, to be able to trap future property additions in case you use the object as a dynamic lookup map. If the object you want to make observable has a regular structure where all members are known up-front, we recommend to use makeObservable as non proxied objects are a little faster, and they are easier to inspect in the debugger and console.log.
```